server/services/inventoryApiService.lua

```diff
@@ -1436,7 +1436,7 @@ function InventoryAPI.openInventory(player, id)
 			triggerAndReloadInventory()
 		else
 			DBService.GetInventory(charid, id, function(inventory)
-				UsersInventories[id][identifier] = createCharacterInventoryFromDB(inventory, identifier)
+				UsersInventories[id][identifier] = createCharacterInventoryFromDB(inventory, charid)
```

Comment:
I mean, you passing the charId here, but it's declared up as the charidentifier of the player who is making the action, the item in the custom inventory could be "owned" by another charId and the update will still fail.

Reply:
Yes you are correct. I think it would be best to just remove the second argument as it is optional and just let it use the previous item owner. I think in this case its a very rare edge case because we are in the branch `not CustomInventoryInfos[id]:isShared()`. So, its a personal/private custom inventory where no other character might have access to. The chance, the charid of items in this inventory are not of the one person that can access it, would be very low. But would be more future-proof to adopt your changes. I will test this on my server and get back to you.